Use Xcode 26 on CI

crazytonyli opened the pull request:
## Description
<!-- Describe the changes, why they are needed, and how they address the issue -->

## Testing instructions
<!-- Consider listing specific testing steps for helping reviewers evaluate the efficacy of the changes -->

<!--

Consider testing the following *yourself* before requesting a review:

- Compatibility with WordPress.com sites and self-hosted Jetpack sites
- Both portrait and landscape orientations
- Light and dark modes
- Dynamic font sizes: larger, smaller and bold text
- High contrast mode
- Screen reader experience (e.g., VoiceOver)
- Languages with large words or with letters/accents not frequently used in English
- Right-to-left languages layout
- Multi-tasking: split view and slide over

-->

crazytonyli committed:
Use Xcode 26

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

crazytonyli committed:
Fix SVProgressHUD

Error: umbrella header for module 'SVProgressHUD' does not include header 'SVIndefiniteAnimatedView.h'

crazytonyli committed:
Update iOS simulator version

dangermattic commented on fastlane/lanes/build.rb:
```diff
@@ -531,4 +533,41 @@ def upload_gutenberg_sourcemaps(sentry_project_slug:, release_version:, build_ve
     )
   end
+
+  def add_photos_to_simulator(device)
```
<table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/WordPress-iOS/blob/9c7d49ab26b476430bfc474e727e1c26a3a5c642/fastlane/lanes/build.rb#L537"></span>Metrics/AbcSize: Assignment Branch Condition size for <code>add_photos_to_simulator</code> is too high. [&lt;9, 19, 8&gt; 22.49/17]</td>
    </tr>
  </tbody>
</table>

crazytonyli committed:
Some UI tests fail on iOS 26

crazytonyli commented on WordPress/UITests/Tests/EditorGutenbergTests.swift:
```diff
@@ -53,6 +53,10 @@ class EditorGutenbergTests_01: EditorGutenbergTests {
+        if #available(iOS 26.0, *) {
+            XCTExpectFailure("No photos in the Photos app")
+        }
+
```
I have created an issue for these test failures: https://linear.app/a8c/issue/CMM-726

crazytonyli committed:
Remove tapping 'ToggleSidebar' button in UI tests

crazytonyli committed:
Fix Reader UI tests

crazytonyli committed:
Replace XCTExpectFailure with XCTSkip
